### PR TITLE
Glimpse

### PR DESCRIPTION
**Pull request by Jetski5822**
Introducing a Glimpse module, this is enabled pertenant.

At the moment the Glimpse code is in flux so *dont merge*

What I need is feedback on some parts.

### CODE REVIEW

File: src/Orchard.Cms.Web/Themes/TheTheme/_ViewImports.cshtml [hunk 1]
```diff
@@ -6,4 +6,5 @@
 @inherits Orchard.DisplayManagement.Razor.RazorPage<TModel>
 @addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
 @addTagHelper *, Orchard.ContentManagement.Display
-@addTagHelper *, Orchard.ResourceManagement
\ No newline at end of file
+@addTagHelper *, Orchard.ResourceManagement
+@addTagHelper *, Glimpse.Agent.AspNet.Mvc
```
Comment (Jetski5822) [inline src/Orchard.Cms.Web/Themes/TheTheme/_ViewImports.cshtml]: @sebastienros Is there a way to dynamically add this?

File: src/Orchard.Data/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -25,6 +27,8 @@ public static IServiceCollection AddDataAccess(this IServiceCollection services)
             services.TryAddDataProvider(name: "Sql Server", value: "SqlConnection", hasConnectionString: true);
             services.TryAddDataProvider(name: "Sql Lite", value: "Sqlite", hasConnectionString: false);
 
+            SqlDialectFactory.RegisterSqlDialect("diagnosticdbconnection", new SqlServerDialect());
```
Comment (Jetski5822) [inline src/Orchard.Data/ServiceCollectionExtensions.cs]: @sebastienros The Dialect stuff in YesSql stops me from wrapping up DBConnections unless I register them. Any ideas?

Comment (sebastienros) [inline src/Orchard.Data/ServiceCollectionExtensions.cs]: I understand the problem. I will look at your code and see what I can change. Right now I think I could add a custom configuration to accept a dialect connection name. Or also provide a way to hook up a connection wrapper which would not alter the dialect. Would that work?

Are there any other changes we should do to make the module not intrusive at all?

File: src/Orchard.Cms.Web/Themes/TheTheme/_ViewImports.cshtml (hunk 1 above)
Comment (Jetski5822) [inline src/Orchard.Cms.Web/Themes/TheTheme/_ViewImports.cshtml]: Sweet- Fixed :)

File: src/Orchard.Data/ServiceCollectionExtensions.cs (hunk 2 above)
Comment (Jetski5822) [inline src/Orchard.Data/ServiceCollectionExtensions.cs]: This is the only one at the moment. The new Glimpse stuff is very limited at the moment, so maybe we could get this stuff in, then I can iterate on it more whilst other people take a look.

**Comment by sebastienros** [comment]
Try to use the REsourceManager in the filter instead of adding a shape which itself need to change the viewimports to add resourcemanager tag helpers.

It's possible that you had to do this because the RM doesn't provide an API to add custom attributes, if this is the case I can do it in a separate PR to unblock you.

**Comment by Jetski5822** [comment]
Got an example?

**Comment by Jetski5822** [comment]
@sebastienros Okay, updated to use the resource manager, can you take a look?

**Comment by sebastienros** [comment]
Not yet perfect. Let me add more things to the RM, you need to be able to create attributes and add "async" which you can't right now. Then you'll just do RM.Include(...). I'll update your or when I have done the changes.

**Comment by Jetski5822** [comment]
@sebastienros did you add those things to the resource manager?

File: src/Orchard.Data/ServiceCollectionExtensions.cs (hunk 2 above)
Comment (Jetski5822) [inline src/Orchard.Data/ServiceCollectionExtensions.cs]: @sebastienros did you get around to doing the dialect stuff?

**Pull request closed without merging**